src/Templates/ModelTemplate.cshtml

```diff
@@ -141,7 +141,7 @@ else
     </text>
 }
 
-@if (Model.BaseIsPolymorphic || Model.IsPolymorphic || Model.AllProperties.Any(p => p.ModelType is DictionaryTypeGo))
+@if (Model.BaseIsPolymorphic || Model.IsPolymorphic || Model.HasFlattenedFields ||  Model.AllProperties.Any(p => p.ModelType is DictionaryTypeGo))
```

Comment:
@mcardosos do you think is any reason to keep this here ?

Reply:
Here you mean in `models.go`?

---
In reply to: [172636578](https://github.com/Azure/autorest.go/pull/103#discussion_r172636578) [](ancestors = 172636578)

Reply:
sorry, somehow posted this. Yes we need this.

I was looking at the old CS where we had tne trick with the alias but we are using this now to populate the discriminator value

Reply:
Question. If model has flattened fields, does it need an unmarshaller? Unmarshaller should only be needed with discriminator stuff

Reply:
yes , is the same issue. if the embedded type has a custom unmarshaller and the embedding type doesnt, the one from the embedded type will be used on the entire embedding type which will result in deserialization errors.

Reply:
:( but we  dont want an unmarshaller just becase it has flattened types, right? only if it is flattened and the flattened stuff has some polymorphism
I would not like to create unmarshallers for everything that has a flattened type

Reply:
correc,t  if we can check that all chain of embedded types does not have any polymorphism than we can have an if on that.  Opened issue to track this https://github.com/Azure/azure-sdk-for-go/issues/1204